macOS builds cannot build _tkinter

Building Python on macOS / OSX succeeds but it can't find tk. The `make` log contains:
```
/usr/include/tk.h:78:11: fatal error: 'X11/Xlib.h' file not found
#       include <X11/Xlib.h>
                ^
1 error generated.
In file included from _imagingtk.c:19:
/usr/include/tk.h:78:11: fatal error: 'X11/Xlib.h' file not found
#       include <X11/Xlib.h>
                ^
```

To fix, `CPPFLAGS` can be set (see https://stackoverflow.com/a/13576206).

--- a/src/commands/install/unix.rs
+++ b/src/commands/install/unix.rs
@@ -77,9 +77,9 @@ pub fn compile_source(
         "--enable-optimizations".to_string(),
     ];
 
-    let mut cflags = Vec::new();
-    let mut cppflags = Vec::new();
-    let mut ldflags = Vec::new();
+    let mut cflags: Vec<String> = Vec::new();
+    let mut cppflags: Vec<String> = Vec::new();
+    let mut ldflags: Vec<String> = Vec::new();
 
     // See https://devguide.python.org/setup/#macos-and-os-x
     #[cfg(target_os = "macos")]

--- a/src/commands/install/unix.rs
+++ b/src/commands/install/unix.rs
@@ -77,6 +77,10 @@ pub fn compile_source(
         "--enable-optimizations".to_string(),
     ];
 
+    let mut cflags = Vec::new();
+    let mut cppflags = Vec::new();
+    let mut ldflags = Vec::new();
+
     // See https://devguide.python.org/setup/#macos-and-os-x
     #[cfg(target_os = "macos")]
     {
@@ -86,8 +90,8 @@ pub fn compile_source(
             let ssl_arg = format!("--with-openssl={}", openssl_prefix);
             configure_args.push(ssl_arg);
         } else {
-            env::set_var("CPPFLAGS", format!("-I{}/include", openssl_prefix));
-            env::set_var("LDFLAGS", format!("-L{}/lib", openssl_prefix));
+            cppflags.push(format!("-I{}/include", openssl_prefix));
+            ldflags.push(format!("-L{}/lib", openssl_prefix));
         };
 
         // Make sure compilation can find zlib
@@ -97,8 +101,12 @@ pub fn compile_source(
             .stdout(Redirection::Pipe)
             .capture()?
             .stdout_str();
-        env::set_var("CFLAGS", format!("-I{}/usr/include", macos_sdk_path.trim()));
+        cflags.push(format!("-I{}/usr/include", macos_sdk_path.trim()));
     }
+
+    env::set_var("CFLAGS", cflags.join(" "));
+    env::set_var("CPPFLAGS", cppflags.join(" "));
+    env::set_var("LDFLAGS", ldflags.join(" "));
 
     let basename = utils::build_basename(&version)?;
     let extract_dir = utils::directory::extracted()?.join(&basename);

--- a/src/commands/install/unix.rs
+++ b/src/commands/install/unix.rs
@@ -102,6 +102,8 @@ pub fn compile_source(
             .capture()?
             .stdout_str();
         cflags.push(format!("-I{}/usr/include", macos_sdk_path.trim()));
+
+        cppflags.push("-I/opt/X11/include".into());
     }
 
     env::set_var("CFLAGS", cflags.join(" "));

